source/includes/_advanced-usage.md

```diff
@@ -104,67 +104,137 @@ __To open the checkout view for a product directly:__
+stateBuilder.buildState({
+    productIds: PRODUCT_IDS,
+    collectionIds: COLLECTION_IDS,
+    shouldFetchStore: IS_SHOP_ON_PAGE,
+})
 ```
 
-Full example:
+This method returns a promise in which the ready state will be available, which should then be passed to `SsrTipserElementsProvider`:
 
 ```typescript
-stateBuilder
-  .buildState(PRODUCT_IDS, COLLECTION_IDS, IS_SHOP_ON_PAGE)
-  .then(initialState => {
-    renderToString(
-      <TipserElementsProvider posId={POS_ID} initialState={initialState}>
-        <YourAppHere />
-      </TipserElementsProvider>
+stateBuilder.buildState({
+    productIds: dataToFetch.productIds,
+    collectionIds: dataToFetch.collectionIds,
+    shouldFetchStore: dataToFetch.shouldFetchStore
+}).then((initialState) => {
+    const markup = renderToString(
+        <TipserElementsProvider posId={POS_ID}>
+            <SsrTipserElementsProvider initialState={initialState}>
+                <App/>
+            </SsrTipserElementsProvider>
+        </TipserElementsProvider>
     );
-  });
+    ...
+ });
 ```
-
-The state should be transferred to the frontend app. You can use the pattern that is known from Redux based apps.
-All you need to do is including the following line in your html response:
+The status should then be transferred to the browser. For this purpose, it can be assigned to the global window object and placed in the returned html document:
 
 ```html
 <script>window.TIPSER_STATE = ${JSON.stringify(initialState)}</script>
 ```
 
-then on the frontend side you are able to reuse it:
+On the client side, use `window.TIPSER_STATE` as the initial State `SsrTipserElementsProvider`:
 
 ```typescript
-hydrate(
-    <TipserElementsProvider posId={POS_ID} initialState={window.TIPSER_STATE}>
-        <YourAppHere />
-    </TipserElementsProvider>,
-    document.getElementById('root')
-);
+<TipserElementsProvider posId={POS_ID}>
+    <SsrTipserElementsProvider initialState={window.TIPSER_STATE}>
+        <App/>
+    </SsrTipserElementsProvider>
+</TipserElementsProvider>
 ```
 
 That's all! The complete example is available [here](https://github.com/Tipser/tipser-elements-ssr-bootstrap).
 
+### Smart method
+In the smart version, the `ComponentsStateSsrManager` class should be used to build the state. For this purpose, you should create an instance of this class, which cannot be global, i.e. a new instance should be created for each request:
+
+```typescript
+const componentsStateSsrManager = new ComponentsStateSsrManager(POS_ID);
+```
+
+Then perform the first rendering that will collect information about the components on the page:
+
+```typescript
+const componentsStateSsrManager = new ComponentsStateSsrManager(POS_ID, 'prod')
+
+const toRender = (
+    <TipserElementsProvider posId={POS_ID}>
+        <SsrTipserElementsProvider componentsStateManager={componentsStateSsrManager}>
+            <App/>
+        </SsrTipserElementsProvider>
+    </TipserElementsProvider>
+);
+
+renderToString(toRender);
```

Comment:
I would put a comment here what is the purpose of this `renderToString()`. It was totally unclear for koket.se when we were discussing this code with them.